docs/nextjs.md

```diff
@@ -0,0 +1,190 @@
+# Use Genkit in a Next.js app 
+
+This page shows how you can use Genkit flows as server actions in your Next.js
+apps.
+
+## Before you begin
+
+You should be familiar with Genkit's concept of [flows](flows), and how to write
+them.
+
+## Create a Next.js project
+
+If you don't already have a Next.js project that you want to add generative AI
+features to, you can create one for the purpose of following along with this
+page:
+
+```posix-terminal
+npx create-next-app@latest
+```
+
+## Install Genkit dependencies
+
+Install the Genkit dependencies into your Next.js app, the same way you would
+for any Genkit project:
+
+1.  Install the core Genkit library:
+
+    ```posix-terminal
+    npm i --save genkit
+    ```
+
+1.  Install at least one model plugin.
+
+    For example, to use Google AI:
+
+    ```posix-terminal
+    npm i --save @genkit-ai/googleai
+    ```
+
+    Or to use Vertex AI:
+
+    ```posix-terminal
+    npm i --save @genkit-ai/vertexai
+    ```
+
+1.  If you didn't install the Genkit CLI globally, you can install it as a
+    development dependency. The tsx tool is also recommended, as it makes
+    testing your code more convenient. Both of these dependencies are optional,
+    however.
+
+    ```posix-terminal
+    npm i --save-dev genkit-cli tsx
+    ```
+
+## Define Genkit flows
+
+Create a new file in your Next.js project to contain your Genkit flows: for
+example, `src/app/genkit.ts`. This file can contain your flows without
+modification; however, because you can only run flows from a server backend, you
+must add the `'use server'` directive to the top of the file.
+
+For example:
+
+```ts
+'use server';
+
+import { gemini15Flash, googleAI } from "@genkit-ai/googleai";
+import { genkit, z } from "genkit";
+
+const ai = genkit({
+  plugins: [googleAI()],
+  model: gemini15Flash,
+});
+
+export const menuSuggestionFlow = ai.defineFlow(
+  {
+    name: "menuSuggestionFlow",
+    inputSchema: z.string(),
+    outputSchema: z.string(),
+  },
+  async (restaurantTheme) => {
+    const { text } = await ai.generate({
+      model: gemini15Flash,
+      prompt: `Invent a menu item for a ${restaurantTheme} themed restaurant.`,
+    });
+    return text;
+  }
+);
+```
+
+## Call your flows 
+
+Now, in your frontend code, you can import your flows and call them as server
+actions.
+
+For example:
+
+```tsx
+'use client';
+
+import { menuSuggestionFlow } from '@/app/genkit';
+import { useState } from 'react';
+
+export default function Home() {
+  const [menuItem, setMenuItem] = useState<string>('');
+
+  async function getMenuItem(formData: FormData) {
+    const theme = formData.get('theme')?.toString() ?? '';
+    const suggestion = await menuSuggestionFlow(theme);
```

Comment:
This actually works???

Reply:
it works locally. should i be more skeptical?

Reply:
No, all good, I was hoping it would, just haven't tested it. 

I'm not sure if it's considered a good practice (re security implications), but still neat.
As a follow up we should also document how to use genkit with API routes....

Reply:
fyi i just successfully deployed it with app hosting